omnibus/files/private-chef-ctl-commands/spec/rotate_credentials_spec.rb

```diff
@@ -0,0 +1,153 @@
+    end
+  end
+
+  context "rotate-credentials" do
+    let(:command) { "rotate-credentials" }
+    let(:params) { ["opscode-solr4"] }
+
+    it_behaves_like "rotation command"
+  end
+
+  context "rotate-all-credentials" do
+    let(:command) { "rotate-all-credentials" }
+    let(:params) { [] }
+
+    it_behaves_like "rotation command"
+  end
+
+  context "rotate-shared-secrets" do
+    let(:command) { "rotate-shared-secrets" }
+    let(:params) { [] }
+
+    it_behaves_like "rotation command"
+  end
+
+  context "require-credential-rotation" do
+    let(:ui) { HighLine.new }
+    let(:prehook_file_path) { "/tmp/prehook" }
+    let(:prehook_file) { StringIO.new }
+
+    before do
+      allow(HighLine).to receive(:new).and_return(ui)
+      allow(subject.ctl).to receive(:run_sv_command).with("stop").and_return(true)
+      allow(subject.ctl).to receive(:get_all_services).and_return([])
+      allow(subject.ctl).to receive(:run_sv_command).with("stop")
+      allow(subject.ctl)
+        .to receive(:credential_prehook_file_path)
+        .and_return(prehook_file_path)
+      allow(File).to receive(:open).with(prehook_file_path, "a+").and_yield(prehook_file)
+    end
+
+    it "bypasses cofirmation if passed --yes" do
+      expect(ui).to_not receive(:ask)
+
+      expect { subject.run_test_omnibus_command("require-credential-rotation", %w{--yes}) }
+        .to raise_error(SystemExit) { |e| expect(e.status).to eq(0) }
+    end
+
+    it "prompts the user for confirmation" do
+      allow(ui).to receive(:ask).and_return("yes")
+      expect(ui).to receive(:ask).with(/confirm/)
+
+      expect { subject.run_test_omnibus_command("require-credential-rotation", []) }
+        .to raise_error(SystemExit) { |e| expect(e.status).to eq(0) }
+    end
+
+    it "exits if the user does not agree" do
+      allow(ui).to receive(:ask).and_return("no")
+      expect(ui).to receive(:ask).with(/confirm/)
+
+      expect { subject.run_test_omnibus_command("require-credential-rotation", []) }
+        .to raise_error(SystemExit) { |e| expect(e.status).to eq(1) }
+    end
+
+    it "stops all services" do
+      expect(subject.ctl).to receive(:run_sv_command).with("stop")
+      expect { subject.run_test_omnibus_command("require-credential-rotation", %w{--yes}) }
+        .to raise_error(SystemExit) { |e| expect(e.status).to eq(0) }
+    end
+
+    it "disables all services" do
+      allow(subject.ctl)
+        .to receive(:get_all_services)
+        .and_return(%w{postgresql oc_erchef})
+      allow(subject.ctl).to receive(:service_enabled?).and_return(true)
+      allow(File).to receive(:unlink).with(/postgresq/).and_return(true)
+      allow(File).to receive(:unlink).with(/oc_erchef/).and_return(true)
+      expect { subject.run_test_omnibus_command("require-credential-rotation", %w{--yes}) }
+        .to raise_error(SystemExit) { |e| expect(e.status).to eq(0) }
+    end
+
+    it "writes the pre-hook lock" do
```

Comment:
Tests seem fine, although it's still hard for me when reading them to devine what the purpose of the command actually is. Ideally I'd be able to tell that from just reading the tests.